### ISSUE TITLE
Fullscreen-ed cards should preserve their original space

### ISSUE THREAD

**Issue by schloerke** [rstudio/bslib#705]
<details>
<summary>Reprex python app</summary>

```python
from __future__ import annotations

import shiny.experimental as x
from shiny import App, ui

piggy_bank = ui.HTML(
    '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" class="bi bi-piggy-bank " style="height:auto;width:100%;fill:currentColor;" aria-hidden="true" role="img" ><path d="M5 6.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0zm1.138-1.496A6.613 6.613 0 0 1 7.964 4.5c.666 0 1.303.097 1.893.273a.5.5 0 0 0 .286-.958A7.602 7.602 0 0 0 7.964 3.5c-.734 0-1.441.103-2.102.292a.5.5 0 1 0 .276.962z"></path>\n<path fill-rule="evenodd" d="M7.964 1.527c-2.977 0-5.571 1.704-6.32 4.125h-.55A1 1 0 0 0 .11 6.824l.254 1.46a1.5 1.5 0 0 0 1.478 1.243h.263c.3.513.688.978 1.145 1.382l-.729 2.477a.5.5 0 0 0 .48.641h2a.5.5 0 0 0 .471-.332l.482-1.351c.635.173 1.31.267 2.011.267.707 0 1.388-.095 2.028-.272l.543 1.372a.5.5 0 0 0 .465.316h2a.5.5 0 0 0 .478-.645l-.761-2.506C13.81 9.895 14.5 8.559 14.5 7.069c0-.145-.007-.29-.02-.431.261-.11.508-.266.705-.444.315.306.815.306.815-.417 0 .223-.5.223-.461-.026a.95.95 0 0 0 .09-.255.7.7 0 0 0-.202-.645.58.58 0 0 0-.707-.098.735.735 0 0 0-.375.562c-.024.243.082.48.32.654a2.112 2.112 0 0 1-.259.153c-.534-2.664-3.284-4.595-6.442-4.595zM2.516 6.26c.455-2.066 2.667-3.733 5.448-3.733 3.146 0 5.536 2.114 5.536 4.542 0 1.254-.624 2.41-1.67 3.248a.5.5 0 0 0-.165.535l.66 2.175h-.985l-.59-1.487a.5.5 0 0 0-.629-.288c-.661.23-1.39.359-2.157.359a6.558 6.558 0 0 1-2.157-.359.5.5 0 0 0-.635.304l-.525 1.471h-.979l.633-2.15a.5.5 0 0 0-.17-.534 4.649 4.649 0 0 1-1.284-1.541.5.5 0 0 0-.446-.275h-.56a.5.5 0 0 1-.492-.414l-.254-1.46h.933a.5.5 0 0 0 .488-.393zm12.621-.857a.565.565 0 0 1-.098.21.704.704 0 0 1-.044-.025c-.146-.09-.157-.175-.152-.223a.236.236 0 0 1 .117-.173c.049-.027.08-.021.113.012a.202.202 0 0 1 .064.199z"></path></svg>'
)
arrow_up = ui.HTML(
    '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" class="bi bi-arrow-up " style="height:1em;width:1em;fill:currentColor;" aria-hidden="true" role="img" ><path fill-rule="evenodd" d="M8 15a.5.5 0 0 0 .5-.5V2.707l3.146 3.147a.5.5 0 0 0 .708-.708l-4-4a.5.5 0 0 0-.708 0l-4 4a.5.5 0 1 0 .708.708L7.5 2.707V14.5a.5.5 0 0 0 .5.5z"></path></svg>'
)

app_ui = ui.page_fluid(
    x.ui.value_box(
        "KPI Title",
        ui.h1(ui.HTML("$1 <i>Billion</i> Dollars")),
        ui.span(arrow_up, " 30% VS PREVIOUS 30 DAYS"),
        showcase=piggy_bank,
        class_="bg-success",
        full_screen=True,
        # showcase_layout=x.ui._valuebox.showcase_left_center(),
    ),
    x.ui.value_box(
        "KPI Title",
        ui.h1(ui.HTML("$1 <i>Billion</i> Dollars")),
        ui.span(arrow_up, " 30% VS PREVIOUS 30 DAYS"),
        showcase=piggy_bank,
        class_="bg-danger",
        full_screen=True,
        showcase_layout=x.ui.showcase_top_right(width="70%"),
    ),
)


app = App(app_ui, server=None)
```

</details>

In this example app, there are two value boxes (which are cards). When the top valuebox is expanded, it will be displayed in front, like a modal. However, it's original footprint has been removed.  While it is _broken_, it will cause things that depending on spacing (such as plots) to re-render.

Ex: The app has two value boxes, a green one and a red one.
![Screenshot 2023-07-20 at 3 26 17 PM](https://github.com/rstudio/bslib/assets/93231/31e4a912-2dd5-441c-a9e1-ff9581e70f65)

When the green valuebox is expanded, the red valuebox moves up to where the green valuebox was located.

![Screenshot 2023-07-20 at 3 26 28 PM](https://github.com/rstudio/bslib/assets/93231/2985fa58-1f94-46b9-a4a3-d4c867e33c57)

I believe the red one should stay where was originally located.

------------------------------

(low priority)

**Comment by gadenbuie** [comment]
Thanks @schloerke. The full-screen card view is a CSS-driven change, so we use `position: absolute` to pop the card out of the document flow and make it full screen. The advantage is that we don't have to move the DOM node around and the mechanism for making the card full screen is quite simple. The downside is that there's a bit of layout shift underneath the expanded card (because we pulled the container out of the document flow).

So, in short, this is expected.
